### DISCUSSION TITLE
How to set Renderable, set Material, apply texture of Nodes

### DISCUSSION THREAD

**Discussion by MukeshBlippar** [sceneview/sceneview#270] [General]
Hi,

I see in SceneView we have ArModelNode, which takes glb as renderable input. But how can I create Basic shapes, apply texture or change material etc.

In old Sceneform we had Material builder & Texture builder to do above all those things. And we can easily set renderable in Node. But how to achieve same in SceneView ?

I have use cases to show image, gif, basic shapes etc in my scene, already I'm using your another Sceneform java library. Planning to migrate.

Would be great if you can share some code sample. And even better if you can add one sample app related tho material & texture.

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, the old Renderable/Material APIs are replaced by a Compose-declarative approach:

**Load a model:**
```kotlin
val model = rememberModelInstance(modelLoader, "models/mymodel.glb")
```

**Access/change materials at runtime:**
```kotlin
val material = model?.materialInstances?.first()
material?.setParameter("baseColorFactor", 1f, 0f, 0f, 1f) // red tint
```

**Create geometry with custom material:**
```kotlin
CubeNode(materialInstance = myMaterial, size = Size(1f))
SphereNode(materialInstance = myMaterial, radius = 0.5f)
```

**ImageNode for textures:**
```kotlin
ImageNode(materialLoader = materialLoader, imageFileLocation = "textures/photo.png")
```

See `samples/android-demo` for working examples.
